osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/infra/kafka/out/GridMeasurementPublishedEventProducer.java

```diff
@@ -8,34 +8,31 @@
 package org.opensmartgridplatform.adapter.kafka.da.infra.kafka.out;
 
 import org.opensmartgridplatform.adapter.kafka.da.application.mapping.DistributionAutomationMapper;
-import org.opensmartgridplatform.adapter.kafka.da.avro.MeterReading;
-import org.opensmartgridplatform.domain.da.measurements.MeasurementReport;
+import org.opensmartgridplatform.adapter.kafka.da.avro.GridMeasurementPublishedEvent;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.kafka.core.KafkaTemplate;
 import org.springframework.stereotype.Service;
 
-// TODO - Rename class once AVRO message format is known
 @Service
-public class MeterReadingProducer {
+public class GridMeasurementPublishedEventProducer {
 
-    private final KafkaTemplate<String, MeterReading> kafkaTemplate;
+    private final KafkaTemplate<String, GridMeasurementPublishedEvent> kafkaTemplate;
 
     private final DistributionAutomationMapper mapper;
 
     @Autowired
-    public MeterReadingProducer(
-            @Qualifier("distributionAutomationKafkaTemplate") final KafkaTemplate<String, MeterReading> kafkaTemplate,
+    public GridMeasurementPublishedEventProducer(
+            @Qualifier("distributionAutomationKafkaTemplate") final KafkaTemplate<String, GridMeasurementPublishedEvent> kafkaTemplate,
             final DistributionAutomationMapper mapper) {
         this.kafkaTemplate = kafkaTemplate;
         this.mapper = mapper;
     }
 
-    public void send(final MeasurementReport measurementReport) {
+    public void send(final String measurement) {
 
-        // TODO - Map measurementReport to correct Avro message format and
-        // send...
-        final MeterReading meterReading = this.mapper.map(measurementReport, MeterReading.class);
+        final GridMeasurementPublishedEvent meterReading = this.mapper.map(measurement,
```

Comment:
With the refactoring of the type from `MeterReading` to `GridMeasurementPublishedEvent`, the name of variable `meterReading` does not look a very good fit anymore.